test/unit/queue/src/QueueTest.cpp

```diff
@@ -65,7 +43,13 @@ TEMPLATE_LIST_TEST_CASE("queueCallbackIsWorking", "[queue]", TestQueues)
 
     alpaka::enqueue(f.m_queue, [&]() { promise.set_value(true); });
 
-    LOOPED_CHECK(30, 100, promise.get_future().get());
+    // A non-blocking queue will always stay empty because the task has been executed immediately.
```

Comment:
I disagree. A non-blocking queue will be non-empty until the background thread has finished executing the task.

Reply:
The background thread is an alpaka workaround which should not affect the behavior of `enqueue`. In a blocking queue, we need the guarantee that if no other thread is running the queue finished all tasks before.